### PR TITLE
Adds program support to the dashboard

### PR DESCRIPTION
**Pull request by jkachel**
(draft - realized that there were a couple bugs while writing this PR)

#### Pre-Flight checklist

- [X] Screenshots and design review for any changes that affect layout or styling
  - [X] Desktop screenshots
- [X] Testing
  - [X] Code is tested
  - [X] Changes have been manually tested

#### What are the relevant tickets?

#725

#### What's this PR do?

Adds support for displaying program enrollments on the dashboard, including enrollments that are implicit. Program enrollments display the program itself with the featured image of the first course, and learners can click to expand a drawer that shows their enrollment within courses in the program.

#### How should this be manually tested?

Enroll in a course that is part of a program. The program should appear on your dashboard. The course should _not_ appear. If your course enrollment is an audit enrollment, the program should be tagged as "Enrolled in free course". If the enrollment is a verified enrollment, the program will have the "Enrolled in certificate course" tag. 

Add another enrollment in a course in the same program. The drawer and course count should reflect the correct number of enrollments. The tags should follow the above test, but if you have an enrollment of each type (i.e. one audit and one verified), you should see both tags. 

Enroll directly within a program (you will have to create a Program Enrollment record manually to do this). The program enrollment should show on the dashboard but it should list no courses, be tagged as "Not enrolled in any courses", and you should get the normal "Enroll Now" card when expanding the drawer.  

Enroll in a course that isn't assigned to a program. The course enrollment should appear on the dashboard as normal.

Modals for course enrollments should work properly regardless of their location (on the dashboard or in the drawer). 

#### Screenshots (if appropriate)
Screen Shot 2022-08-02 at 8 10 54 AM
Screen Shot 2022-08-02 at 8 11 09 AM
Screen Shot 2022-08-02 at 8 11 16 AM

Width for the drawer is 1280px/701px = 55%; can be changed if necessary.

### CODE REVIEW

File: frontend/public/src/containers/pages/DashboardPage.js [hunk 1]
```diff
@@ -234,14 +251,50 @@ export class DashboardPage extends React.Component<
...
+
+    if (programEnrollments) {
+      const enrollmentMatches = programEnrollments.map(programEnrollment => {
+        return programEnrollment.enrollments.some(elem => elem.id === enrollment.id)
+      })
+
+      showCard = !enrollmentMatches.includes(true)
+    }
+
+    if (showCard) {
```
Comment (annagav) [inline frontend/public/src/containers/pages/DashboardPage.js:273]: Could you invert this if and return null here instead, the less number of lines we have inside the if the better :)

**Review by annagav — CHANGES_REQUESTED** [review]
Code looks good, just a minor comment. Going to test functionality next

File: frontend/public/src/containers/pages/DashboardPage.js (hunk 1 above)
Comment (jkachel) [reply frontend/public/src/containers/pages/DashboardPage.js]: Should be done in https://github.com/mitodl/mitxonline/pull/766/commits/358db0bd1fb222d962f565f31d6441be790eb3a1

**Review by annagav — COMMENTED** [review]
The UI looks good. One thing I could see is the blu tag `Program` on the program card. Did we decide to omit that?

**Comment by jkachel** [comment]
Forgot about Program tag - added it in https://github.com/mitodl/mitxonline/pull/766/commits/c8661cab935ab51492966f03dcc7a9eca47e2e09 (and rebased from main).

**Review by annagav — APPROVED** [review]
👍

**Pull request merged**